src/Mayaqua/Network.c

```diff
@@ -5809,14 +5809,53 @@ SOCK *ListenAnyPortEx2(bool local_only, bool disable_ca)
 	return NULL;
 }
 
-int cb_test(int a, X509_STORE_CTX *ctx)
+// Verify client SSL certificate during TLS handshake.
+//
+// (actually, only save the certificate for later authentication in Protocol.c)
+int SslCertVerifyCallback(int preverify_ok, X509_STORE_CTX *ctx)
 {
-	WHERE;
-	return 1;
+	SSL *ssl;
+	struct SslClientCertInfo *clientcert;
+
+	ssl = X509_STORE_CTX_get_ex_data(ctx, SSL_get_ex_data_X509_STORE_CTX_idx());
+	clientcert = SSL_get_ex_data(ssl, GetSslClientCertIndex());
+
+	if (clientcert != NULL)
+	{
+		clientcert->PreverifyOk = (bool)preverify_ok
;
```

Comment:
Different line break (**CRLF** instead of **LF**).

Reply:
I would write an explicit cast:
```c++
clientcert->PreverifyOk = static_cast<bool>(preverify_ok);
```

Reply:
Fixed the line break, but static_cast would not work here, because Network.c is not complied as a C++ source file:
```
...
cc -DNDEBUG -DVPN_SPEED -DUNIX -DUNIX_LINUX -DCPU_64 -D_REENTRANT -DREENTRANT -D_THREAD_SAFE -D_THREADSAFE -DTHREAD_SAFE -DTHREADSAFE -D_FILE_OFFSET_BITS=64 -I./src/ -I./src/Cedar/ -I./src/Mayaqua/
 -O2 -fsigned-char -m64 -c src/Mayaqua/Network.c -o tmp/objs/Mayaqua/Network.o
src/Mayaqua/Network.c: In function 'SslCertVerifyCallback':
src/Mayaqua/Network.c:5825:29: error: 'static_cast' undeclared (first use in this function)
   clientcert->PreverifyOk = static_cast<bool>(preverify_ok);
```

Reply:
Removed this line altogether, since, as you correctly noted, the value assigned to PreverifyOk is never used.

Reply:
Oh, I didn't notice that it isn't a C++ file, sorry.